Caching: Added new picture and tweaked text (#1723)
HTTPArchive/almanac.httparchive.org: src/content/en/2020/caching.md

raghuramakrishnan71_bio: Enterprise architect at <a href="https://www.tcs.com/">Tata Consultancy Services</a>, working on large digital transformation programs in the public sector. A technology enthusiast with a special interest in performance engineering. An avid traveler, intrigued by astronomy, history, biology, and advancements in medicine. A strong follower of the 47th verse, Chapter 2 of Bhagavad Gita "karmaṇy-evādhikāras te mā phaleṣhu kadāchana" meaning "You have a right to perform your prescribed duty, but you are not entitled to the fruits of action."

discuss: 2056
results: https://docs.google.com/spreadsheets/d/1fYmpSN3diOiFrscS75NsjfsrKXzxxhUMNcYSqXnQJQU/
queries: 20_Caching

> featured_quote: Caching provides a significant performance benefit by avoiding costly network requests - it helps both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
> featured_stat_1: 25.6%
> featured_stat_label_1: Percentage of HTTP responses don't include any caching information, which means they're probably being cached wrongly
> featured_stat_2: 21.4%
> featured_stat_label_2: Percentage of responses that aren't using revalidation, resulting in wasted bandwidth and increased latency
> featured_stat_3: 78.6%
> featured_stat_label_3: Percentage of sites could save up to 2MB of bandwidth by improving their caching policy

Reviewer: This is what it looks like:

![image](https://user-images.githubusercontent.com/10931297/101818511-e0680000-3b1b-11eb-9856-e341b5f6510c.png)

I think a bit of a mouthful to be honest! Difficult to digest those stats quickly. Also some repetition in there (do we need to say it's a percentage when they have a percentage sign?)

Finally I think the 3rd stat is the wrong way of looking at this! Up to 2Mb of saving could mean 0.0001 Mb of saving. Let's flip it around and look at the ones over 2Mb instead of under (I also think we should change the chapter text to say this too).

```suggestion
featured_quote: Caching provides a significant performance benefit by avoiding costly network requests - it helps both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
featured_stat_1: 25.6%
featured_stat_label_1: HTTP responses with no caching information
featured_stat_2: 21.4%
featured_stat_label_2: Responses that cannot be revalidated
featured_stat_3: 21.3%
featured_stat_label_3: Sites that could save over 2MB on repeat visits with better caching
```

This looks like this, which I think is better:

![image](https://user-images.githubusercontent.com/10931297/101819654-76e8f100-3b1d-11eb-896d-67e63ef58c8f.png)

So I also think you should change this line in the text:

> Lighthouse also indicates how many bytes could be saved on repeat views by enabling a longer cache policy. Of the sites that could benefit from additional caching, 78.6% of them can reduce their page weight by up to 2MB!

to this:

> Lighthouse also indicates how many bytes could be saved on repeat views by enabling a longer cache policy. Of the sites that could benefit from additional caching, 21.3% of them can reduce their page weight by over 2MB!

> reviewers: [csswizardry, jzyang, jaisanth, Soham-S-Sarkar]

Reviewer: Did csswizardy contribute? I ping him earlier and he wasn't sure he had!

Reviewer 2: IIUC @jzyang was the only reviewer? If so let's update to this.
```suggestion
reviewers: [jzyang]
```

analysts: [raghuramakrishnan71]
translators: []

> roryhewitt_bio: Enterprise Architect at <a href="https://www.akamai.com/">Akamai</a>, who is passionate about performance. A British ex-patriate, he has lived in San Francisco for more than twenty years. In his spare time, he's a long-distance adventure motorcyclist, competition-level snowboarder and boxer with multiple black belts in various styles of karate. He likes being known as a troublemaker. Most importantly, he's a father and husband and the owner of Luna the cat. Find him at <a href="https://www.linkedin.com/in/roryhewitt/">LinkedIn</a>.

Reviewer 2: FYI you could also add your LinkedIn profile to your [contributor config](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/config/2020.json#L1131)

raghuramakrishnan71_bio: Enterprise architect at <a href="https://www.tcs.com/">Tata Consultancy Services</a>, working on large digital transformation programs in the public sector. A technology enthusiast with a special interest in performance engineering. An avid traveler, intrigued by astronomy, history, biology, and advancements in medicine. A strong follower of the 47th verse, Chapter 2 of Bhagavad Gita "karmaṇy-evādhikāras te mā phaleṣhu kadāchana" meaning "You have a right to perform your prescribed duty, but you are not entitled to the fruits of action."

discuss: 2056
results: https://docs.google.com/spreadsheets/d/1fYmpSN3diOiFrscS75NsjfsrKXzxxhUMNcYSqXnQJQU/
queries: 20_Caching

> #featured_quote: Caching benefits both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
> #featured_stat_1: 25.6
> #featured_stat_label_1: Number of HTTP responses don't include any caching information, which means they're probably being cached wrongly
> #featured_stat_2: 21.4%
> #featured_stat_label_2: Number of responses that aren't using revalidation, resulting in wasted bandwidth and increased latency
> #featured_stat_3: 78.6%
> #featured_stat_label_3: Number of sites could save up to 2MB of bandwidth by improving their caching policy

Reviewer: Can activate these now:

```suggestion
featured_quote: Caching benefits both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
featured_stat_1: 25.6%
featured_stat_label_1: Number of HTTP responses don't include any caching information, which means they're probably being cached wrongly
featured_stat_2: 21.4%
featured_stat_label_2: Number of responses that aren't using revalidation, resulting in wasted bandwidth and increased latency
featured_stat_3: 78.6%
featured_stat_label_3: Number of sites could save up to 2MB of bandwidth by improving their caching policy
```

Also this is what it looks like now:

![Featured quote](https://user-images.githubusercontent.com/10931297/101750638-3494c500-3ac7-11eb-96e1-b099ffe871dc.png)

The quotes itself is quite short compared to the labels.

Reviewer 2: I'd also suggest rewording "Number" to "Percent"

> #featured_stat_label_3: Percentage of sites could save up to 2MB of bandwidth by improving their caching policy

Reviewer 2: Reapplying @bazzadp's suggested changes
```suggestion
featured_quote: Caching provides a significant performance benefit by avoiding costly network requests - it helps both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
featured_stat_1: 25.6%
featured_stat_label_1: Percentage of HTTP responses don't include any caching information, which means they're probably being cached wrongly
featured_stat_2: 21.4%
featured_stat_label_2: Percentage of responses that aren't using revalidation, resulting in wasted bandwidth and increased latency
featured_stat_3: 78.6%
featured_stat_label_3: Percentage of sites could save up to 2MB of bandwidth by improving their caching policy
```

results: https://docs.google.com/spreadsheets/d/1fYmpSN3diOiFrscS75NsjfsrKXzxxhUMNcYSqXnQJQU/
queries: 20_Caching
#featured_quote: Caching benefits both the end users (they get their web pages quickly) and the companies serving the web pages (reducing the load on their servers). Caching really is a win-win!
#featured_stat_1: **One quarter** of all HTTP responses don't include any caching information, which means they're probably being cached wrongly
#featured_stat_label_1: TODO
#featured_stat_2: More than **one fifth** of all responses aren't using revalidation, resulting in wasted bandwidth and increased latency
#featured_stat_label_2: TODO

> #featured_stat_3: Our chapter has a cat picture!

Reviewer 2: 😸 Can we use a real stat here instead?

Author: We could. I was trying to add some humor :) I'll dig out a useful one...

Author: @rviscomi what should go in the `featured_stat_label` fields?

Reviewer 2: Here's an example of how it looks on the home page

![image](https://user-images.githubusercontent.com/1120896/101710365-522e4400-3a5f-11eb-9000-dab2ee8d0359.png)

The big text like 72.58% is the stat and "percent of length values..." is the label.

Author: Ah - ok then, I'll change it - currently I have the `stat` just the text string. More commits!

Author: Also, it looks like there's an error trying to compress at least one of the images - probably the new `luna-cat.png` image... Any thoughts on what I should do? I can make it smaller, if that helps - right now it's about 2MB.

* An example of static content is an image.  For instance, a picture of a cat is the same regardless of who's requesting it or where the requester is located.
* An example of dynamic content is a list of events which are specific to a geographic location. The list will be different based on the requester's location.

{{ figure_markup(

>   image="luna-cat.png",

Reviewer 2: Is this really worth the 2.19 MB?

>   link="https://almanac.httparchive.org/static/images/2020/caching/luna-cat.png",

Reviewer: The link happens automatically so no need to include this. It's meant for if you want a different link for some reason.

```suggestion
```
